system-parachains/asset-hubs/asset-hub-kusama/src/treasury.rs

```diff
@@ -37,14 +44,64 @@ pub type TreasuryPaymaster = parachains_common::pay::LocalPay<
 	xcm_config::LocationToAccountId,
 >;
 
+#[derive(
+	Default,
+	MaxEncodedLen,
+	Encode,
+	Decode,
+	DecodeWithMemTracking,
+	TypeInfo,
+	Clone,
+	Eq,
+	PartialEq,
+	Debug,
+)]
+pub struct BurnDestinationAccount(pub Option<polkadot_core_primitives::AccountId>);
+
+impl BurnDestinationAccount {
+	pub fn is_set(&self) -> bool {
+		self.0.is_some()
+	}
+}
+
+pub type BalancesNegativeImbalance = <Balances as Currency<AccountId>>::NegativeImbalance;
+pub struct TreasuryBurnHandler;
+
+impl OnUnbalanced<BalancesNegativeImbalance> for TreasuryBurnHandler {
+	fn on_nonzero_unbalanced(amount: BalancesNegativeImbalance) {
+		let destination = dynamic_params::treasury::BurnDestination::get();
+
+		if let BurnDestinationAccount(Some(account)) = destination {
+			// Must resolve into existing but better to be safe.
+			Balances::resolve_creating(&account, amount);
+		} else {
+			// If no account to destinate the funds to, just drop the
+			// imbalance.
+			<() as OnUnbalanced<_>>::on_nonzero_unbalanced(amount)
+		}
+	}
+}
+
+impl Get<Permill> for TreasuryBurnHandler {
+	fn get() -> Permill {
+		let destination = dynamic_params::treasury::BurnDestination::get();
+
+		if destination.is_set() {
```

Comment:
Not sure if i understand this: Why is it only respecting the BurnPortion when there is a destination set?  
Then it should be a tuple and not two independent parameters.

Reply:
@pandres95 Wdyt as the original author of https://github.com/polkadot-fellows/runtimes/pull/511? Follow-up changing this to tuple?

Reply:
For now, parameters in the Relay Chain are empty, so there might not be a need for a data migration.

<img width="1712" height="411" alt="Screenshot 2025-09-04 at 11 10 14 am" src="https://github.com/user-attachments/assets/8b1bcab1-9e59-4d6e-b6c9-705fdd37f540" />